File: backend/voting-app/src/main/java/com/tu/votingapp/utils/mappers/MunicipalityMapper.java
package com.tu.votingapp.utils.mappers;

import com.tu.votingapp.dto.general.MunicipalityDTO;
import com.tu.votingapp.entities.MunicipalityEntity;
import org.mapstruct.Mapper;

@Mapper(componentModel = "spring", uses = {RegionMapper.class})
public interface MunicipalityMapper {
    MunicipalityDTO toDto(MunicipalityEntity entity);
    MunicipalityEntity toEntity(MunicipalityDTO dto);
}

File: frontend/android/app/src/main/java/com/example/android/utils/mappers/MunicipalityMapper.kt
package com.example.android.utils.mappers

import com.example.android.dto.response.MunicipalityResponseDTO
import com.example.android.entity.MunicipalityEntity
import com.example.android.entity.RegionEntity
import com.example.android.utils.mappers.RegionMapper.toDto

object MunicipalityMapper {
    fun MunicipalityResponseDTO.toEntity(): MunicipalityEntity = MunicipalityEntity(
        id = this.id,
        name = this.name,
        population = this.population,
        regionId = region.id
    )

    fun MunicipalityEntity.toDto(region: RegionEntity): MunicipalityResponseDTO = MunicipalityResponseDTO(
        id = this.id!!,
        name = this.name,
        population = this.population,
        region = region.toDto()
    )
}